CMakeLists.txt

```diff
@@ -505,6 +505,11 @@ seastar_generate_protobuf (
   IN_FILE ${CMAKE_CURRENT_SOURCE_DIR}/src/proto/metrics2.proto
   OUT_DIR ${Seastar_GEN_BINARY_DIR}/src/proto)
 
+set(FORTIFY_BUILD_TYPES "Release;RelWithDebInfo;Dev")
+if(CMAKE_BUILD_TYPE IN_LIST FORTIFY_BUILD_TYPES)
+  add_definitions(-D_FORTIFY_SOURCE=2)
+endif()
```

Comment:
This is added in a second commit that we can drop. It's only real purpose here is to demonstrate that the tests pass with `_FORTIFY_SOURCE=2` when it's undefined on `src/core/thread.cc`.

Reply:
this works if we use a single-configuration generator. but it does not work if we use a multi-config generator to generate the rules. if we use 
```
cmake -G "Ninja Multi-Config" 
```
to generate `build.ninja`, we cannot get the "current" build mode via `CMAKE_BUILD_TYPE` anymore. as this variable is determined at the configure time. but with multi-config generator, one can pick which configuration at build time, like:
```
cmake --build build --config Dev --target seastar
```

so, i'd suggest use the generator expression, which is expanded to different value specific to each build configuration, so we could use something like:
```cmake
target_compile_definitions (seastar
  PRIVATE
    $<$<IN_LIST:$<CONFIG>,${FORTIFY_BUILD_TYPES}>:_FORTIFY_SOURCE=2)
```

in which,

1. use `PRIVATE` so we don't force the parent project's to use the same fortification level as seastar does
2. instead of using `add_definitions()` use `target_compile_definitions()`, more specific this way
3. use generator expression as explained above.

Reply:
@tchaikov i'm not suggesting that we leave this here, it is merely for testing. do you want fortify sources enabled?